### PR TITLE
Set default absolute/relative paths for new projects 

### PR DESCRIPTION
**Pull request by mhugent**
Pathes in the QGIS project can be set to 'absolute' or 'relative'. It is however not possible to set a default for newly created projects (new projects are always with relative paths by default). This PR adds an option to set the default for new projects to absolute or relative paths. Like this, it is possible to set the default for new projects to absolute paths (e.g. for the use in organisations where they have geodata on network shares but move around the project files).
What are your thoughts about it?

### CODE REVIEW

File: src/ui/qgsoptionsbase.ui [hunk 1]
```diff
@@ -1037,7 +1037,45 @@
...
+                      <property name="text">
+                       <string>Default paths</string>
+                      </property>
+                     </widget>
+                    </item>
+                    <item>
+                     <widget class="QComboBox" name="mDefaultPathsComboBox">
+                      <item>
+                       <property name="text">
+                        <string>relative</string>
```
Comment (nyalldawson) [inline src/ui/qgsoptionsbase.ui:1053]: "Relative" (HIG guideline is title case for combobox entries)

File: src/ui/qgsoptionsbase.ui [hunk 2]
```diff
@@ -1037,7 +1037,45 @@
...
+                       </property>
+                      </item>
+                      <item>
+                       <property name="text">
+                        <string>absolute</string>
```
Comment (nyalldawson) [inline src/ui/qgsoptionsbase.ui:1058]: Absolute

File: tests/src/core/testqgsproject.cpp [hunk 3]
```diff
@@ -699,5 +700,24 @@ void TestQgsProject::testCrsExpressions()
...
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), true );
+  QgsProject p1;
+  bool p1PathsAbsolute = p1.readBoolEntry( QStringLiteral( "Paths" ), QStringLiteral( "/Absolute" ), false );
+
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), false );
+  QgsProject p2;
+  bool p2PathsAbsolute = p2.readBoolEntry( QStringLiteral( "Paths" ), QStringLiteral( "/Absolute" ), false );
+
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), bk_defaultRelativePaths );
+
```
Comment (nyalldawson) [inline tests/src/core/testqgsproject.cpp:717]: I'd suggest also testing this with an explicit QgsProject::clear call here too, to mimic what qgis app actually does

File: src/core/project/qgsproject.cpp [hunk 4]
```diff
@@ -843,7 +843,9 @@ void QgsProject::clear()
   // XXX THESE SHOULD BE MOVED TO STATUSBAR RELATED SOURCE
   writeEntry( QStringLiteral( "PositionPrecision" ), QStringLiteral( "/Automatic" ), true );
   writeEntry( QStringLiteral( "PositionPrecision" ), QStringLiteral( "/DecimalPlaces" ), 2 );
-  writeEntry( QStringLiteral( "Paths" ), QStringLiteral( "/Absolute" ), false );
+
+  bool defaultRelativePaths = mSettings.value( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), true ).toBool();
```
Comment (nyalldawson) [inline src/core/project/qgsproject.cpp:847]: would be good to add this to https://github.com/qgis/QGIS/blob/master/resources/qgis_global_settings.ini too to make it easier for enterprise deployments to set

**Comment by mhugent** [comment]
Thanks for the feedback @nyalldawson

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**